node/coinstacks/solana/api/src/utils.ts

```diff
@@ -0,0 +1,59 @@
+import { EnrichedTransaction } from 'helius-sdk'
+import { Tx } from './models'
+import axios from 'axios'
+import { handleError } from '@shapeshiftoss/common-api'
+import { RPC_API_KEY, INDEXER_URL } from './constants'
+import axiosRetry, { isNetworkOrIdempotentRequestError } from 'axios-retry'
+
+export const axiosNoRetry = axios.create({ timeout: 5000, params: { 'api-key': RPC_API_KEY } })
+const axiosWithRetry = axios.create({ timeout: 10000, params: { 'api-key': RPC_API_KEY } })
+
+axiosRetry(axiosWithRetry, {
+  shouldResetTimeout: true,
+  retries: 5,
+  retryDelay: (retryCount, err) => {
+    console.log({ retryCount, err })
+    // don't add delay on top of request timeout
+    if (err.code === 'ECONNABORTED') return 0
+    // add exponential delay for network errors
+    return axiosRetry.exponentialDelay(retryCount, undefined, 500)
+  },
+  retryCondition: (err) =>
+    isNetworkOrIdempotentRequestError(err) ||
+    (!!err.response && err.response.status >= 400 && err.response.status < 600),
+})
+
+const exponentialDelay = async (retryCount: number) =>
+  new Promise((resolve) => setTimeout(resolve, axiosRetry.exponentialDelay(retryCount, undefined, 500)))
```

Comment:
could be worth pulling this out into `common/api/utils.ts`

Reply:
Done in [37ba9ed](https://github.com/shapeshift/unchained/pull/1053/commits/37ba9edb1be37c7338ff96166dc24614031fbd65)

Reply:
yeah saw that, could make a package, but this is good for now. thanks!